js/controllers/todos_controller.js

```diff
@@ -1,4 +1,125 @@
-App.TodosController = Ember.ObjectController.extend({
+
+Abs.TasksController = Ember.ObjectController.extend({
     actions: {
+        editTasks: function () {
+            this.set('editMode', true);
+        },
+        changeTask: function () {
+            var me = this;
+            Ember.run.later((function () {
+                me.set('editMode', false);
+                if (Ember.isEmpty(me.get('model.title'))) {
+                    me.send('removeTask');
+                } else {
+                    me.get('model').save();
+                }
+            }), 100);
+        },
+        removeTask: function () {
+            var Task = this.get('model');
+            Task.deleteRecord();
+            Task.save();
+        }
+    },
+
+    editMode: false,
+
+    isDone: function (key, value) {
+        var model = this.get('model');
+
+        if (value === undefined) {
+            // property being used as a getter
+            return model.get('isDone');
+        } else {
+            // property being used as  setter
+            model.set('isDone', value);
+            model.save();
+            return value;
+        }
+    }.property('model.isDone')
+});
+
+Abs.AbsController = Ember.ArrayController.extend({
```

Comment:
I know its better to create new file , but I wanted to keep your files and don't delete or add a file, because I thought you certain files for me to edit.   


     On Monday, September 28, 2015 4:41 PM, Ruan Naude <notifications@github.com> wrote:
   

 In js/controllers/todos_controller.js:> +    isDone: function (key, value) {
> +        var model = this.get('model');
> +
> +        if (value === undefined) {
> +            // property being used as a getter
> +            return model.get('isDone');
> +        } else {
> +            // property being used as  setter
> +            model.set('isDone', value);
> +            model.save();
> +            return value;
> +        }
> +    }.property('model.isDone')
> +});
> +
> +Abs.AbsController = Ember.ArrayController.extend({
Why did you create another controller in the same file?—
Reply to this email directly or view it on GitHub.